src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -254,7 +254,7 @@ public synchronized boolean append(TxnHeader hdr, Record txn)
         for (File f : files) {
             long fzxid = Util.getZxidFromName(f.getName(), LOG_FILE_PREFIX);
             if (fzxid > snapshotZxid) {
-                continue;
+                break;
             }
             // the files
```

Comment:
@a470577391 Yes,since the files is already order by zxid asc.
I have two suggestions personally if you want to open a new JIRA:
1: rename the method to a explicit  one, `getLogFiles` is ambiguous
2. use Java8 lambda to rewrite the method elegantly,`stream, filter, max `will be useful.

Reply:
Thank you for your suggestion.
1. i'm going to try.
2. i think use Java8 may be unfriendly to the lower version.

Reply:
2. ZooKeeper runs in Java, release 1.8 or greater :smile:

Reply:
@a470577391 ZooKeeper 3.4.x runs on Java7, version 3.5.x and 3.6.x are not stable yet, they use Java8.